### PR TITLE
Feature: Retry rejected transactions

### LINKED ISSUE #354: Feature: Retry rejected transactions
https://www.notion.so/efdn/Feature-Retry-transaction-79251f0e8ed64985ac584c81604c26a8
image

After rejecting a tx for whatever reason, a nice-to-have feature would be to have a "retry" button available within the contribution modal.

### PR DESCRIPTION
**Pull request by corwintines**
Fixes: https://github.com/ethereum/clrfund/issues/354

**What Changed**
- Functionality added in Transaction component to optionally show a retry button with a retry function
- ContributionModal and ReallocationModal taking advantage of new retry button

### CODE REVIEW

File: vue-app/src/components/Transaction.vue [hunk 1]
```diff
@@ -2,13 +2,22 @@
...
+          v-if="displayCloseBtn"
+          class="btn-secondary close-btn"
+          @click="$emit('close')"
+        >
+          Close
+        </button>
+        <button
+          v-if="displayRetryBtn"
+          class="btn-primary close-btn"
+          @click="retryMethod()"
```
Comment (pettinarip) [inline vue-app/src/components/Transaction.vue]: You can use the builtin event props in this case and we could have a cleaner interface in this component (as we are doing with the close event in the close button).
We could do `@click="$emit('retry')"` here, and from the consumer:
```
<transaction @retry="doSomething">
```

More about this here: https://vuejs.org/v2/guide/components-custom-events.html

**Review by pettinarip — CHANGES_REQUESTED** [review]
Looks great. Just a small refactor to have a cleaner component interface.

File: vue-app/src/components/Transaction.vue (hunk 1 above)
Comment (corwintines) [reply vue-app/src/components/Transaction.vue]: Cool!

**Review by pettinarip — APPROVED** [review]
GJ!

**Review by samajammin — APPROVED** [review]
👍

**Pull request merged**